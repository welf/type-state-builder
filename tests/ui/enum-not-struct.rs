use type_state_builder::TypeStateBuilder;

// This should be an error: enums are not supported, only structs
#[derive(TypeStateBuilder)]
enum NotAStruct {
    Variant1(String),
    Variant2 { field: u32 },
}

fn main() {
    let _instance = NotAStruct::builder()
        .build();
}