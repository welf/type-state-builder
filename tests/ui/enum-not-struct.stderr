error: TypeStateBuilder only supports structs, not enums
 --> tests/ui/enum-not-struct.rs:5:1
  |
5 | / enum NotAStruct {
6 | |     Variant1(String),
7 | |     Variant2 { field: u32 },
8 | | }
  | |_^
